### PR TITLE
[SPARK-29603][YARN] Support application priority for YARN priority scheduling

### PR DESCRIPTION
**Pull request by yaooqinn**
### What changes were proposed in this pull request?

Priority for YARN to define pending applications ordering policy, those with higher priority have a better opportunity to be activated. YARN CapacityScheduler only.

### Why are the changes needed?

Ordering pending spark apps
### Does this PR introduce any user-facing change?

add a conf
### How was this patch tested?

add ut

### CODE REVIEW

**Comment by yaooqinn** [comment]
cc @jerryshao @vanzin @cloud-fan, thanks in advance

**Comment by abellina** [comment]
@yaooqinn Might point out in the config documentation that FIFO policy is required for priority (it is default in CapacityScheduler, but could be set to FAIR).

File: docs/running-on-yarn.md [hunk 1]
```diff
@@ -419,6 +419,15 @@ To use a custom metrics.properties for the application master and executors, upd
   in YARN ApplicationReports, which can be used for filtering when querying YARN apps.
   </td>
 </tr>
+<tr>
+  <td><code>spark.yarn.app.priority</code></td>
```
Comment (vanzin) [inline docs/running-on-yarn.md]: We don't have any configs that use `.app`, so I'd call this just `spark.yarn.priority`.

File: docs/running-on-yarn.md [hunk 2]
```diff
@@ -419,6 +419,15 @@ To use a custom metrics.properties for the application master and executors, upd
...
+  <td>(none)</td>
+  <td>
+  Application priority for YARN to define pending applications ordering policy, those with higher
+  integer value have a better opportunity to be activated. Currently Application priority is supported
```
Comment (vanzin) [inline docs/running-on-yarn.md]: "Currently, YARN only supports application priority when using FIFO ordering policy."

File: resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala [hunk 3]
```diff
@@ -307,6 +307,12 @@ private[spark] class Client(
       }
     }
     appContext.setUnmanagedAM(isClientUnmanagedAMEnabled)
+
+    sparkConf.get(APPLICATION_PRIORITY) match {
+      case Some(v) => appContext.setPriority(Priority.newInstance(v))
+      case None => logDebug(s"${APPLICATION_PRIORITY.key} is not set. " +
```
Comment (vanzin) [inline resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala]: This debug message is not really useful. Just do `sparkConf.get(...).foreach`.

Comment (yaooqinn) [reply resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala]: done, thanks

**Comment by vanzin** [comment]
Merging to master.

**Pull request closed without merging**